app/views/admin/roles/_form.html.erb

```diff
@@ -5,48 +5,116 @@
 
 <%= form_for role, as: :role, url: role_url_for(role) do |form| %>
   <%= form.errors %>
+    <%= render "govuk_publishing_components/components/input", {
+        label: {
+          text: "Role title"
+        },
+        value: role.name,
+        name: "role[name]",
+        id: "role_name",
+        type: "text"}
+    %>
 
-  <fieldset>
-    <%= form.text_field :name, label_text: 'Role title' %>
+    <%= render "govuk_publishing_components/components/select", {
+      id: "role_type_id",
```

Comment:
The error from the error summary banner tries to link to `role_type` so we need this to be `id: "role_type"`

Reply:
cool, thank you. Yes, think I understand error handling better now.